File: appsets-android-native/appsets/src/main/kotlin/xcj/app/appsets/ui/compose/LocalCompositions.kt
package xcj.app.appsets.ui.compose

import androidx.activity.OnBackPressedDispatcher
import androidx.compose.runtime.staticCompositionLocalOf

val LocalBackPressedDispatcher =
    staticCompositionLocalOf<OnBackPressedDispatcher> { error("No Back Dispatcher provided") }


//提供所有需要登录拦截页面名
val LocalPageRouteNameNeedLoggedProvider = staticCompositionLocalOf<MutableList<String>> {
    mutableListOf(
        PageRouteNameProvider.ConversationOverviewPage,
        PageRouteNameProvider.ConversationDetailsPage,
        PageRouteNameProvider.GroupConversationPage,
        PageRouteNameProvider.UserProfilePage,
        PageRouteNameProvider.AddScreenPostPage,
        PageRouteNameProvider.CreateAppPage,
    )
}

File: AppSets-iOS/AppSets/com/app/appsets/ui/swiftui/LocalCompositions.swift
//
//  LocalStatic.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/11.
//

import Foundation

let LocalPageRouteNameNeedLoggedProvider:StaticProvider<Array<String>> = staticProvider {
    [
        PageRouteNameProvider.ConversationOverviewPage,
        PageRouteNameProvider.ConversationDetailsPage,
        PageRouteNameProvider.UserProfilePage
    ]
}
